File: Construct3/Construct3/Library/Telerik.OpenAccess.SDK/IDEIntegrations/templates/newWinFormReport.vb
Public Partial Class %ClassName%
    Inherits System.Windows.Forms.Form

    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        'Add any initialization after the InitializeComponent() call

    End Sub


    Private Sub objectView1_PositionChanged()
        Me.labelNavLocation.Text = (((Me.BindingContext(ObjectView1).Position + 1).ToString + " of  ") _
                    + Me.BindingContext(ObjectView1).Count.ToString)
    End Sub
    Private Sub buttonNavFirst_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonNavFirst.Click
        Me.BindingContext(ObjectView1).Position = 0
        Me.objectView1_PositionChanged()
    End Sub

    Private Sub buttonNavPrev_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonNavPrev.Click
        Me.BindingContext(ObjectView1).Position = (Me.BindingContext(ObjectView1).Position - 1)
        Me.objectView1_PositionChanged()
    End Sub

    Private Sub buttonNavNext_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonNavNext.Click
        Me.BindingContext(ObjectView1).Position = (Me.BindingContext(ObjectView1).Position + 1)
        Me.objectView1_PositionChanged()
    End Sub

    Private Sub buttonLast_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonLast.Click
        Me.BindingContext(ObjectView1).Position = (Me.ObjectView1.Count - 1)
        Me.objectView1_PositionChanged()
    End Sub

    Private Sub buttonAdd_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonAdd.Click
        Try
            'Clear out the current edits
            Me.BindingContext(ObjectView1).EndCurrentEdit()
            Me.BindingContext(ObjectView1).AddNew()
        Catch eEndEdit As System.Exception
            System.Windows.Forms.MessageBox.Show(eEndEdit.Message)
        End Try
        Me.objectView1_PositionChanged()
    End Sub

    Private Sub buttonDelete_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonDelete.Click
        If (Me.BindingContext(ObjectView1).Count > 0) Then
            Me.BindingContext(ObjectView1).RemoveAt(Me.BindingContext(ObjectView1).Position)
            Me.objectView1_PositionChanged()
        End If
    End Sub

    Private Sub buttonCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonCancel.Click
        Me.BindingContext(ObjectView1).CancelCurrentEdit()
        Me.ObjectProvider1.CancelAll()
    End Sub

    Private Sub buttonSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonSave.Click
        Me.BindingContext(ObjectView1).EndCurrentEdit()
        Me.ObjectProvider1.SaveAll()
    End Sub

    Private Sub Form3_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.objectView1_PositionChanged()
    End Sub

End Class

File: Construct3/Construct3/Library/Telerik.OpenAccess.SDK/IDEIntegrations/templates/newWinFormReport.cs
//Copyright (c) %CompanyName%. All rights reserved.
//
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace %DefaultNamespace%
{
	/// <summary>
	/// Summary description for %ClassName%.
	/// </summary>
	public partial class %ClassName% : System.Windows.Forms.Form
	{
		public %ClassName%()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// Add any constructor code after InitializeComponent call
			//
			// Add the following code if you want to dispose on close
			this.Closed += new System.EventHandler(this.FormClosedEvent);
			this.objectView1_PositionChanged();
		}


		private void FormClosedEvent(object sender, System.EventArgs e)
		{
			this.Dispose();
		}

		private void objectView1_PositionChanged()
		{
			this.labelNavLocation.Text = ((((this.BindingContext[objectView1].Position + 1)).ToString() + " of  ") 
				+ this.BindingContext[objectView1].Count.ToString());
		}

		private void buttonNavFirst_Click(object sender, System.EventArgs e)
		{
			this.BindingContext[objectView1].Position = 0;
			this.objectView1_PositionChanged();
		}

		private void buttonNavNext_Click(object sender, System.EventArgs e)
		{
			this.BindingContext[objectView1].Position = (this.BindingContext[objectView1].Position + 1);
			this.objectView1_PositionChanged();
		}

		private void buttonNavPrev_Click(object sender, System.EventArgs e)
		{
			this.BindingContext[objectView1].Position = (this.BindingContext[objectView1].Position - 1);
			this.objectView1_PositionChanged();
		}

		private void buttonLast_Click(object sender, System.EventArgs e)
		{
			this.BindingContext[objectView1].Position = (this.objectView1.Count - 1);
			this.objectView1_PositionChanged();
		}

		private void buttonAdd_Click(object sender, System.EventArgs e)
		{
			try 
			{
				// Clear out the current edits
				this.BindingContext[objectView1].EndCurrentEdit();
				this.BindingContext[objectView1].AddNew();
			}
			catch (System.Exception eEndEdit) 
			{
				System.Windows.Forms.MessageBox.Show(eEndEdit.Message);
			}
			this.objectView1_PositionChanged();
		}

		private void buttonDelete_Click(object sender, System.EventArgs e)
		{
			if ((this.BindingContext[objectView1].Count > 0)) 
			{
				DialogResult r = MessageBox.Show("Do you really want to delete it ?","confirm", MessageBoxButtons.YesNo );
				if (r == DialogResult.Yes)
				{
					this.BindingContext[objectView1].RemoveAt(this.BindingContext[objectView1].Position);
					this.objectView1_PositionChanged();
				}
			}
		}

		private void buttonSave_Click(object sender, System.EventArgs e)
		{
			this.BindingContext[objectView1].EndCurrentEdit();
			this.objectProvider1.SaveAll();
			this.objectView1_PositionChanged();
		}

		private void buttonCancel_Click(object sender, System.EventArgs e)
		{
			DialogResult r = MessageBox.Show("Do you really want to cancel ?","confirm", MessageBoxButtons.YesNo);
			if (r == DialogResult.Yes)
			{
				this.BindingContext[objectView1].CancelCurrentEdit();
				this.objectProvider1.CancelAll();
				this.objectView1_PositionChanged();
			}
		}

	}
}
